pkg/ccl/backupccl/show.go

```diff
@@ -420,62 +395,71 @@ you must pass the 'encryption_info_dir' parameter that points to the directory o
 			} else {
 				return err
 			}
-		} else if len(incLocations) > 0 {
-			// There was no error, so check for incrementals.
-			//
-			// If there are incrementals, find the backup paths and return them with the store.
-			// Otherwise, use the vacuous placeholders above.
-
-			incStore, err = p.ExecCfg().DistSQLSrv.ExternalStorageFromURI(ctx, incLocations[0], p.User())
-			if err != nil {
-				return err
-			}
-			incPaths, err = FindPriorBackups(ctx, incStore, IncludeManifest)
-			if err != nil {
-				return errors.Wrapf(err, "make incremental storage")
-			}
-			manifestDirs = getManifestDirs(computedSubdir, incLocations[0], incPaths, explicitIncPaths)
 		}
 		mem := p.ExecCfg().RootMemoryMonitor.MakeBoundAccount()
 		defer mem.Close(ctx)
 
-		return infoReader.showBackup(ctx, &mem, store, incStore, encryption, incPaths,
-			manifestDirs, resultsCh)
+		var (
+			info        backupInfo
+			memReserved int64
+		)
+		info.subdir = computedSubdir
+
+		mkStore := p.ExecCfg().DistSQLSrv.ExternalStorageFromURI
+
+		info.defaultURIs, info.manifests, info.localityInfo, memReserved,
+			err = resolveBackupManifests(
+			ctx, &mem, baseStores, mkStore, fullyResolvedDest,
+			fullyResolvedIncrementalsDirectory, hlc.Timestamp{}, encryption, p.User())
+		defer func() {
+			mem.Shrink(ctx, memReserved)
+		}()
+		// If backup is locality aware, check that user passed at least some localities
+		// TODO (msbutler, benbardin): this is an extremely crude check that the user is passing at
+		// least as many URIS as there are localities in the backup.
+		// This check is only meant for the 22.1 backport. Ben is working on a much more robust check
+		for _, locMap := range info.localityInfo {
+			if len(locMap.URIsByOriginalLocalityKV) > len(dest) && subdir != "" {
+				p.BufferClientNotice(ctx, pgnotice.Newf("Your backup contains %d localities; however, "+
```

Comment:
I'd like to keep this as a warning for 22.1 because an error would break the following user flow: 
1) user runs a locality aware backup
2) user runs SHOW BACKUP by only passing the default URI.

Reply:
👍